### ISSUE TITLE
E1507 (invalid-envvar-value) does not accept `builtins.str` as a valid type for `os.getenv` argument

### ISSUE THREAD

**Issue by CCardosoDev** [PyCQA/pylint#5091]
### Bug description

```python
If the input argument of `os.getenv` is of type `builtins.str` pylint complains with the error message `os.getenv does not support builtins.str type argument (invalid-envvar-value)`. This should not be the case since `builtins.str` is a string. Which is the expected type like mentioned [here](https://vald-phoenix.github.io/pylint-errors/plerr/errors/stdlib/E1507) "Env manipulation functions support only string type arguments.".

This happens when the key value of the environment variable comes from a data class attribute. Example:


@dataclass
class EnvVar:
    """Environment variable to be set."""

    name: str
    _initial_value: Optional[str] = None

    def __post_init__(self):
        """Save current env var value in `self._initial_value`."""

        self._initial_value = os.getenv(self.name)
```

And the warning is related to `os.getenv(self.name)`.
```


### Configuration

_No response_

### Command used

```shell
pylint env.py
```

### Pylint output

```shell
`env.py:11:30 E1507 os.getenv does not support builtins.str type argument (invalid-envvar-value)`
```

### Expected behavior

No warning raised when the input argument of `os.getenv` is `builtins.str`

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.4
```

### OS / Environment

mascOS Big Sur (Versin 11.5.2)
